spyder/plugins/layout/plugin.py

```diff
@@ -0,0 +1,792 @@
+
+    def get_layout(self, layout_id):
+        """
+        Get a registered layout by his ID.
+
+        Parameters
+        ----------
+        layout_id : string
+            The ID of the layout.
+
+        Returns
+        -------
+        Instance of a spyder.plugins.layout.api.BaseGridLayoutType subclass
+            Layout.
+
+        """
+        return self.get_container().get_layout(layout_id)
+
+    def setup_layout(self, default=False):
+        """Initialize mainwindow layout."""
+        prefix = 'window' + '/'
+        settings = self.load_window_settings(prefix, default)
+        hexstate = settings[0]
+
+        self._first_spyder_run = False
+        if hexstate is None:
+            # First Spyder execution:
+            self.main.setWindowState(Qt.WindowMaximized)
+            self._first_spyder_run = True
+            self.setup_default_layouts('default', settings)
+
+            # Now that the initial setup is done, copy the window settings,
+            # except for the hexstate in the quick layouts sections for the
+            # default layouts.
+            # Order and name of the default layouts is found in config.py
+            section = 'quick_layouts'
+            get_func = self.get_conf
+            order = get_func(section, 'order')
+
+            # Restore the original defaults if reset layouts is called
+            if default:
+                self.set_conf(section, 'active', order)
+                self.set_conf(section, 'order', order)
+                self.set_conf(section, 'names', order)
+
+            for index, _name, in enumerate(order):
+                prefix = 'layout_{0}/'.format(index)
+                self.save_current_window_settings(prefix, section,
+                                                  none_state=True)
+
+            # Store the initial layout as the default in spyder
+            prefix = 'layout_default/'
+            section = 'quick_layouts'
+            self.save_current_window_settings(prefix, section, none_state=True)
+            self._current_quick_layout = 'default'
+
+        self.set_window_settings(*settings)
+
+    def setup_default_layouts(self, index, settings):
+        """Setup default layouts when run for the first time."""
+        main = self.main
+        main.setUpdatesEnabled(False)
+
+        first_spyder_run = bool(self._first_spyder_run)  # Store copy
+
+        if first_spyder_run:
+            self.set_window_settings(*settings)
+        else:
+            if self._last_plugin:
+                if self._last_plugin._ismaximized:
+                    self.maximize_dockwidget(restore=True)
+
+            if not (main.isMaximized() or self._maximized_flag):
+                main.showMaximized()
+
+            min_width = main.minimumWidth()
+            max_width = main.maximumWidth()
+            base_width = main.width()
+            main.setFixedWidth(base_width)
+
+        # IMPORTANT: order has to be the same as defined in the config file

```

Comment:
Was this preserved with the migration? Jus checking.

Reply:
Yep, I think so :+1: